README.md
# nature-is-amazing README

This is the README for your extension "nature-is-amazing". After writing up a brief description, we recommend including the following sections.

test

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

- `myExtension.enable`: enable/disable this extension
- `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

---

## Working with Markdown

**Note:** You can author your README using Visual Studio Code. Here are some useful editor keyboard shortcuts:

- Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux)
- Toggle preview (`Shift+CMD+V` on macOS or `Shift+Ctrl+V` on Windows and Linux)
- Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (macOS) to see a list of Markdown snippets

### For more information

- [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
- [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**


CHANGELOG.md
# Change Log

All notable changes to the "nature-is-amazing" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

.github/solution.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import * as cheerio from 'cheerio';
import axios from 'axios';

export function activate(context: vscode.ExtensionContext) {
  // Use the console to output diagnostic information (console.log) and errors (console.error)
  // This line of code will only be executed once when your extension is activated
  console.log(
    'Congratulations, your extension "nature-is-amazing" is now active!'
  );

  // The command has been defined in the package.json file
  // Now provide the implementation of the command with registerCommand
  // The commandId parameter must match the command field in package.json
  let disposable = vscode.commands.registerCommand(
    'extension.makeMeHappy',
    async () => {
      // The code you place here will be executed every time your command is executed

      // on Startup open a new gif

      // scroll through loaded gifs https://github.com/microsoft/vscode-extension-samples/tree/master/comment-sample

      // when input make me happy internatilised siehe  https://github.com/microsoft/vscode-extension-samples/tree/master/i18n-sample

      // display progressbar https://github.com/microsoft/vscode-extension-samples/tree/master/progress-sample

      // 1. open new window to load the gif

      // 2. crawl random twitter https://twitter.com/natureamazin

      // 3. Display the tweet inside

      // Display a message box to the user

      vscode.window.showInformationMessage('Make me Happy triggered!');

      // Create and show panel
      const panel = vscode.window.createWebviewPanel(
        'makeMeHappy',
        'Make me Happy',
        vscode.ViewColumn.One,
        {
          // Enable scripts in the webview
          enableScripts: true,
        }
      );

      const tweetIDs = await fetchTweetIDs();

      if (Array.isArray(tweetIDs)) {
        const tweetID = getRandomTweetId(tweetIDs);
        panel.webview.html = getMakeMeHappyTweet(tweetID);
      }

      // And set its HTML content
    }
  );

  context.subscriptions.push(disposable);
}

function getRandomTweetId(tweetIDs: Array<string>) {
  return tweetIDs[Math.floor(Math.random() * tweetIDs.length)];
}

function fetchTweetIDs() {
  const TWITTER_USER_ENDPOINT = 'https://twitter.com/natureamazin';

  return axios.get(TWITTER_USER_ENDPOINT).then((response: any) => {
    if (response.status === 200) {
      const html = response.data;
      const $ = cheerio.load(html);
      const tweetIds = $('li.stream-item')
        .map((index, elem) => {
          return $(elem).data('itemId');
        })
        .get();
      return tweetIds;
    }
  });
}

function getMakeMeHappyTweet(tweetId: string) {
  return ` 
  <!DOCTYPE html>
  <html lang="en">
  <head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Cat Coding</title>
  </head>
  <body>
  <iframe border=0 frameborder=0 height=800 width=600 src="https://twitframe.com/show?url=${encodeURIComponent(
    'https://twitter.com/natureamazin/status/'
  ) + tweetId}"></iframe>  </body>
  </html>`;
}

// this method is called when your extension is deactivated
export function deactivate() {}


src/extension.ts
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';
import * as cheerio from 'cheerio';
import axios from 'axios';

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
  // Use the console to output diagnostic information (console.log) and errors (console.error)
  // This line of code will only be executed once when your extension is activated
  console.log(
    'Congratulations, your extension "nature-is-amazing" is now active!'
  );

  // The command has been defined in the package.json file
  // Now provide the implementation of the command with registerCommand
  // The commandId parameter must match the command field in package.json
  let disposable = vscode.commands.registerCommand(
    'extension.makeMeHappy',
    async () => {
      // The code you place here will be executed every time your command is executed

      // on Startup open a new gif

      // scroll through loaded gifs https://github.com/microsoft/vscode-extension-samples/tree/master/comment-sample

      // when input make me happy internatilised siehe  https://github.com/microsoft/vscode-extension-samples/tree/master/i18n-sample

      // 1. open new window to load the gif

      // 2. crawl random twitter https://twitter.com/natureamazin and display progressbar https://github.com/microsoft/vscode-extension-samples/tree/master/progress-sample

      // Display a message box to the user

      vscode.window.showInformationMessage('Make me Happy triggered!');

      // Create and show panel
      const panel = vscode.window.createWebviewPanel(
        'makeMeHappy',
        'Make me Happy',
        vscode.ViewColumn.One,
        {
          // Enable scripts in the webview
          enableScripts: true,
        }
      );

      const tweetIDs = await fetchTweetIDs();
      if (Array.isArray(tweetIDs)) {
        const tweetID = getRandomTweetId(tweetIDs);
        panel.webview.html = getMakeMeHappyTweet(tweetID);
      }

      // And set its HTML content
    }
  );

  context.subscriptions.push(disposable);
}

function getRandomTweetId(tweetIDs: Array<string>) {
  return tweetIDs[Math.floor(Math.random() * tweetIDs.length)];
}

function fetchTweetIDs() {
  const TWITTER_USER_ENDPOINT = 'https://twitter.com/natureamazin';

  return axios.get(TWITTER_USER_ENDPOINT).then((response: any) => {
    if (response.status === 200) {
      const html = response.data;
      const $ = cheerio.load(html);
      const tweetIds = $('li.stream-item')
        .map((index, elem) => {
          return $(elem).data('itemId');
        })
        .get();
      return tweetIds;
    }
  });
}

function getMakeMeHappyTweet(tweetId: string) {
  return ` 
  <!DOCTYPE html>
  <html lang="en">
  <head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Cat Coding</title>
  </head>
  <body>
  <iframe border=0 frameborder=0 height=800 width=600 src="https://twitframe.com/show?url=${encodeURIComponent(
    'https://twitter.com/natureamazin/status/'
  ) + tweetId}"></iframe>  </body>
  </html>`;
}

// this method is called when your extension is deactivated
export function deactivate() {}
